File: core/src/main/java/com/lyl/core/mvpbase/BaseActivity.java
package com.lyl.core.mvpbase;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

/**
 * Create By: lyl
 * Date: 2019-11-28 09:51
 */

public abstract class BaseActivity<V extends BaseView, T extends BasePresenter<V>> extends AppCompatActivity {
    protected T presenter;

    protected abstract T createPresenter();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        presenter = createPresenter();
        if (presenter != null) {
            presenter.attachView((V) this);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        presenter.detachView();
    }
}

File: app/src/main/java/com/lyl/mvptest/mvp/huanfu/BaseActivity.kt
package com.lyl.mvptest.mvp.huanfu

import android.app.Activity
import android.os.Bundle
import androidx.core.view.LayoutInflaterCompat

/**
 * User: lyl
 * Date: 2019-07-27 10:17
 */
open class BaseActivity : Activity() {

    var skinFactory: SkinFactory? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        SkinManager.getSkinManager().setContext(this@BaseActivity)
        skinFactory = SkinFactory()
        LayoutInflaterCompat.setFactory2(layoutInflater, skinFactory!!)
    }

    public fun apply() {
        skinFactory!!.apply()
    }

    override fun onResume() {
        super.onResume()
        skinFactory!!.apply()
    }
}